### ISSUE TITLE
EnTK on localhost

### ISSUE THREAD

**Issue by mpbl** [radical-cybertools/radical.entk#191]
Hi,

It seems that when running on localhost, something is not finishing in the EnTK.
All tasks get computed properly to the end, data are staged out according to the setup.

Heartbeats keep being sent long after the walltime specified in the resource description has passed. The end of the debug output is:
```
2017-12-13 15:37:34,423: radical.entk.wfprocessor: wfprocessor                     : MainThread     : INFO    : Terminating enqueue-thread
2017-12-13 15:37:34,423: radical.entk.appmanager: MainProcess                     : synchronizer-thread: INFO    : Received radical.entk.pipeline.0000 with state DONE
2017-12-13 15:37:37,004: radical.entk.task_manager: MainProcess                     : heartbeat      : INFO    : Received heartbeat response
2017-12-13 15:37:37,006: radical.entk.task_manager: MainProcess                     : heartbeat      : INFO    : Sent heartbeat request
2017-12-13 15:37:37,006: radical.entk.task_manager: task-manager                    : MainThread     : INFO    : Received heartbeat request
2017-12-13 15:37:37,007: radical.entk.task_manager: task-manager                    : MainThread     : INFO    : Sent heartbeat response
2017-12-13 15:37:47,008: radical.entk.task_manager: MainProcess                     : heartbeat      : INFO    : Received heartbeat response
2017-12-13 15:37:47,009: radical.entk.task_manager: MainProcess                     : heartbeat      : INFO    : Sent heartbeat request
2017-12-13 15:37:47,010: radical.entk.task_manager: task-manager                    : MainThread     : INFO    : Received heartbeat request
2017-12-13 15:37:47,010: radical.entk.task_manager: task-manager                    : MainThread     : INFO    : Sent heartbeat response
.... and then the heartbeat thing goes forever
```

The example I am running is (private repo, but Vivek has access to it):
https://bitbucket.org/mpbl/simpy/src/7216f7bd2caf714ac3c54ebd2e092edd1bdcf489/examples/forward_local/?at=entk_integration

There is no emergency on our side, as real cases will have to go through a scheduler anyway and Vivek has shown it was working well in these cases.

Let me know if you need more information or if I am doing something wrong.

**Comment by vivek-bala** [comment]
Hey Matthieu, can you print the branch and commit of EnTK that you are using?

**Comment by mpbl** [comment]
Hi, Vivek,

Here is what ```git log``` gives me: 
```
commit f76cddfc17298c03634472e8e75b391aead518dc (HEAD -> devel, origin/devel)
Author: Vivekanandan (Vivek) Balasubramanian <vivek.balasubramanian@rutgers.edu>
Date:   Tue Nov 14 13:41:39 2017 -0500

    Update README.md
```

Let me try again with a newer version, as https://github.com/radical-cybertools/radical.entk/commit/221251f5d6173bc2325599afe55917a1a3054fc7 might have improved this.

**Comment by mpbl** [comment]
Hi, Vivek,
I have go the same issue with the latest commits from the devel branch.

**Comment by vivek-bala** [comment]
Thanks for checking. Let me test with a smaller example and get back to you.

**Comment by mpbl** [comment]
Hi, Vivek,

Should I retry with the devel branches of saga, rp & entk? Or do I need to checkout other branches?
Thanks,

Matthieu

**Comment by vivek-bala** [comment]
Hey Matthieu! Unless you were using a specific branch originally, you can use the rp version from pypi (which will install saga from pypi). The fix has been made in the "rc/0.6" branch of EnTK, so please use that branch of EnTK (regardless of branch you used originally). 

Thanks for testing!

**Comment by vivek-bala** [comment]
Closing for now. Please reopen if this is still an issue.
